libs/scripts/personalization.js

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable no-console */
+let utils;
+
+const CLASS_EL_DELETE = 'p13n-deleted';
+const CLASS_EL_REPLACE = 'p13n-replaced';
+
+export const PERSONALIZATION_TAGS = {
+  chrome: () => navigator.userAgent.includes('Chrome') && !navigator.userAgent.includes('Mobile'),
+  firefox: () => navigator.userAgent.includes('Firefox') && !navigator.userAgent.includes('Mobile'),
+  android: () => navigator.userAgent.includes('Android'),
+  ios: () => /iPad|iPhone|iPod/.test(navigator.userAgent),
+  loggedout: () => !window.adobeIMS?.isSignedInUser(),
+  loggedin: () => window.adobeIMS?.isSignedInUser(),
+  darkmode: () => window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches,
+  lightmode: () => !PERSONALIZATION_TAGS.darkmode(),
+};
+
+// Replace any non-alpha chars except comma, space and hyphen
+const RE_KEY_REPLACE = /[^a-z0-9\- ,=]/g;
+const RE_SPACE_COMMA = /[ ,]/;
+
+const MANIFEST_KEYS = [
+  'action',
+  'selector',
+  'pagefilter',
+  'page filter',
+  'page filter optional',
+];
+
+const createFrag = (url) => {
+  const a = utils.createTag('a', { href: url }, url);
+  const p = utils.createTag('p', undefined, a);
+  return p;
+};
+
+const COMMANDS = {
+  insertcontentafter: (el, target) => el.insertAdjacentElement('afterend', createFrag(target)),
+  insertcontentbefore: (el, target) => el.insertAdjacentElement('beforebegin', createFrag(target)),
+  removecontent: (el, target) => (target !== 'false' && el.classList.add(CLASS_EL_DELETE)),
+  replacecontent: (el, target) => {
+    if (el.classList.contains(CLASS_EL_REPLACE)) return;
+    el.insertAdjacentElement('beforebegin', createFrag(target));
+    el.classList.add(CLASS_EL_DELETE, CLASS_EL_REPLACE);
+  },
+};
+
+const VALID_COMMANDS = Object.keys(COMMANDS);
+
+const GLOBAL_CMDS = [
+  'insertscript',
+  'replacefragment',
+  'replacepage',
+  'updatemetadata',
```

Comment:
maybe we can use camel case

Reply:
everything is lowercased so that user capitalization doesn't matter